src/Illuminate/Queue/DatabaseQueue.php

```diff
@@ -291,13 +291,11 @@ protected function markJobAsReserved($job)
      */
     public function deleteReserved($queue, $id)
     {
-        $this->database->beginTransaction();
-
-        if ($this->database->table($this->table)->lockForUpdate()->find($id)) {
-            $this->database->table($this->table)->where('id', $id)->delete();
-        }
-
-        $this->database->commit();
+        $this->database->transaction(function () use ($queue, $id) {
+            if ($this->database->table($this->table)->lockForUpdate()->find($id)) {
+                $this->database->table($this->table)->where('id', $id)->delete();
+            }
+        }, 5);
```

Comment:
This hard coding of `5` seems strange. At the minimum it probably should be a configurable number pulled from the DB queue config? It can be defaulted to `5` there?